providers/common/provider_seeding.c

```diff
@@ -23,7 +23,7 @@ int ossl_prov_seeding_from_dispatch(const OSSL_DISPATCH *fns)
          * multiple versions of libcrypto (e.g. one static and one dynamic), but
          * sharing a single fips.so. We do a simple sanity check here.
          */
-#define set_func(c, f) if (c == NULL) c = f; else if (c != f) return 0;
+#define set_func(c, f) if (c == NULL) c = f; else if (c != f) return 0
```

Comment:
FWIW: This is what I was taught how to do it when I was a C beginner:
```suggestion
#define set_func(c, f) do { if (c == NULL) c = f; else if (c != f) return 0; } while (0)
```

Reply:
> use a do{}while(0) loop

Sure.  I mis-read the "else if" so the dangling else issue still exists.  Done.  Also added an undef.  Updated commit pushed.